dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEventAnalyticsManager.java

```diff
@@ -115,16 +116,26 @@
     public JdbcEventAnalyticsManager( JdbcTemplate jdbcTemplate, StatementBuilder statementBuilder,
         ProgramIndicatorService programIndicatorService,
         ProgramIndicatorSubqueryBuilder programIndicatorSubqueryBuilder,
-        EventTimeFieldSqlRenderer timeFieldSqlRenderer )
+        EventTimeFieldSqlRenderer timeFieldSqlRenderer, ExecutionPlanCache executionPlanCache )
     {
-        super( jdbcTemplate, statementBuilder, programIndicatorService, programIndicatorSubqueryBuilder );
+        super( jdbcTemplate, statementBuilder, programIndicatorService, programIndicatorSubqueryBuilder,
+            executionPlanCache );
         this.timeFieldSqlRenderer = timeFieldSqlRenderer;
     }
 
     @Override
     public Grid getEvents( EventQueryParams params, Grid grid, int maxLimit )
     {
-        withExceptionHandling( () -> getEvents( params, grid, getEventsOrEnrollmentsSql( params, maxLimit ) ) );
+        String sql = getEventsOrEnrollmentsSql( params, maxLimit );
+
+        if ( params.analyzeOnly() )
+        {
+            executionPlanCache.addExecutionPlan( params.getAnalyzeOrderId(), sql );
```

Comment:
Yes we need it. Grid object (instance) is in "http request scope". To avoid the situation I have to pass the grid ref to all methods (there are many) on the way down to the object responsible for sql string creation, I am using the @Service RequestExecutionPlanCache and let the IoC container inject it where it is needed. The best case will be define the scope of injection as a Request. This unfortunatelly does not work, cause within HttpRequest we are using various Tasks running in various Threads, which break the scope consistency(the problem is the request scope is bounded to LocalThread object). That's why the Service must be a singleton. AnalyzeOrderId allows me to locate sql queries belong to the related http request.

Reply:
done

Reply:
discussed in huddle